ui/tests/acceptance/sync/secrets/overview-test.js

```diff
@@ -39,4 +39,35 @@ module('Acceptance | sync | destination', function (hooks) {
     await click(ts.overview.table.action('details'));
     assert.dom(ts.tab('Secrets')).hasClass('active', 'Navigates to secrets view for destination');
   });
+
+  test('it should show opt-in banner and modal if secrets-sync is not activated and clear it when opt-in confirmed', async function (assert) {
+    assert.expect(5);
+    server.get('/sys/activation-flags', () => {
+      assert.ok(true, 'Request on initial load to check if secrets-sync is activated');
+      return {
+        data: {
+          activated: [''],
+          unactivated: ['secrets-sync'],
+        },
+      };
+    });
+    await click(ts.navLink('Secrets Sync'));
+    assert.dom(ts.overview.optInBanner).exists('Opt-in banner is shown');
+    await click(ts.overview.optInBannerEnable);
+    assert.dom(ts.overview.optInModal).exists('Opt-in modal is shown');
+    assert.dom(ts.overview.optInConfirm).isDisabled('Confirm button is disabled when checkbox is unchecked');
+    server.get('/sys/activation-flags', () => {
+      return {
+        data: {
+          activated: ['secrets-sync'],
+          unactivated: [],
+        },
+      };
+    });
+    await click(ts.overview.optInCheck);
+    await click(ts.overview.optInConfirm);
+    // Locally, the banner disappears, but on CI we need a route transition, which is why I do not have an assertion inside the activation-flags get handler because the assert count is not the same locally vs. in CI.
+    await click(ts.navLink('Secrets Sync'));
+    assert.dom(ts.overview.optInBanner).doesNotExist('Opt-in banner does not show after confirmation');
```

Comment:
Since we're not actually mocking real world functionality here and changing the endpoint on confirm I think this assertion and the "activated" endpoint on line 59 could be a separate test. More granular tests are easier to debug, and it makes sense to isolate that to just assert that the activated endpoint does not render the banner.

I think it'd be more useful here to assert that clicking "confirm" hits the expected endpoint, which we can do by adding the route to the top of the test (`server.post('/sys/activation-flags/secrets-sync/activate'`) with an assertion inside.